src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        threshold = (int) (capacity * LOAD_FACTOR);
+        this.table = (Node<K, V>[]) new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        int hashValue = hash(key);
+        int index = findIndex(hashValue);
+        if (table[index] == null) {
+            table[index] = new Node<>(hashValue, key, value, null);
+        } else {
+            Node<K, V> iterator = table[index];
+            while (true) {
+                if (iterator.key == key || Objects.equals(iterator.key, key)) {
```

Comment:
```suggestion
                if (Objects.equals(iterator.key, key) {
```
Let's make it easier. Under kapot method "equlas" checks arguments for null.

Reply:
It is true. Thank you.